In [2]:
import requests
import os 
import pandas as pd 
from dotenv import load_dotenv
import json
import time 


In [3]:
import sys
print("Python path:", sys.executable)

Python path: c:\Users\willo\miniconda3\envs\lol-env\python.exe


In [4]:
load_dotenv()

True

In [5]:
riot_key = os.getenv('RIOT_API_KEY')



In [6]:
account_want = "BANGBANG"
acc_tag = "7999"

# for now ill use americas but we could always change it to be any region
url = f"https://americas.api.riotgames.com/riot/account/v1/accounts/by-riot-id/{account_want}/{acc_tag}?"

headers = {
    "X-Riot-Token": riot_key,           
    "User-Agent": "My League App 1.0"   
}

response = requests.get(url,headers=headers)


In [7]:
basic_info = response.json()

basic_info

{'puuid': 'U3fXW6HC3JpGV72px0i9_-yUAnXCClhqkhAUcatcwtso_1lzP9tp2sWU3RLL3Yugf4IuWbxJ-peSEA',
 'gameName': 'BANGBANG',
 'tagLine': '7999'}

In [8]:
puuid = basic_info['puuid'] 

spec_summoner = f"https://na1.api.riotgames.com/lol/summoner/v4/summoners/by-puuid/{puuid}"

summoner_response = requests.get(spec_summoner,headers=headers)


summoner_response.json()

{'puuid': 'U3fXW6HC3JpGV72px0i9_-yUAnXCClhqkhAUcatcwtso_1lzP9tp2sWU3RLL3Yugf4IuWbxJ-peSEA',
 'profileIconId': 6852,
 'revisionDate': 1754974543521,
 'summonerLevel': 320}

In [9]:
matchurl = f"https://americas.api.riotgames.com/lol/match/v5/matches/by-puuid/{puuid}/ids?start=0&count=20&api_key={riot_key}"

matches_response = requests.get(matchurl,headers=headers)

matchees = matches_response.json()

In [10]:
match_we_want = matchees[0]
spec_match = f"https://americas.api.riotgames.com/lol/match/v5/matches/{match_we_want}?api_key={riot_key}"

the_match = requests.get(spec_match,headers=headers)

specific_match = the_match.json()

In [11]:
specific_match['info']['participants'][0]

{'PlayerScore0': 0,
 'PlayerScore1': 0,
 'PlayerScore10': 0,
 'PlayerScore11': 0,
 'PlayerScore2': 0,
 'PlayerScore3': 0,
 'PlayerScore4': 0,
 'PlayerScore5': 0,
 'PlayerScore6': 0,
 'PlayerScore7': 0,
 'PlayerScore8': 0,
 'PlayerScore9': 0,
 'allInPings': 0,
 'assistMePings': 0,
 'assists': 26,
 'baronKills': 0,
 'basicPings': 0,
 'challenges': {'12AssistStreakCount': 0,
  'HealFromMapSources': 3118.4344782829285,
  'InfernalScalePickup': 0,
  'SWARM_DefeatAatrox': 0,
  'SWARM_DefeatBriar': 0,
  'SWARM_DefeatMiniBosses': 0,
  'SWARM_EvolveWeapon': 0,
  'SWARM_Have3Passives': 0,
  'SWARM_KillEnemy': 0,
  'SWARM_PickupGold': 0,
  'SWARM_ReachLevel50': 0,
  'SWARM_Survive15Min': 0,
  'SWARM_WinWith5EvolvedWeapons': 0,
  'abilityUses': 110,
  'acesBefore15Minutes': 1,
  'alliedJungleMonsterKills': 0,
  'baronTakedowns': 0,
  'blastConeOppositeOpponentCount': 0,
  'bountyGold': 392.1710205078125,
  'buffsStolen': 0,
  'completeSupportQuestInTime': 0,
  'controlWardsPlaced': 0,
  'damagePer

In [12]:
specific_match['metadata']['participants'].index(puuid)

8

In [13]:
specific_match['info']['participants'][2]

{'PlayerScore0': 0,
 'PlayerScore1': 0,
 'PlayerScore10': 0,
 'PlayerScore11': 0,
 'PlayerScore2': 0,
 'PlayerScore3': 0,
 'PlayerScore4': 0,
 'PlayerScore5': 0,
 'PlayerScore6': 0,
 'PlayerScore7': 0,
 'PlayerScore8': 0,
 'PlayerScore9': 0,
 'allInPings': 0,
 'assistMePings': 0,
 'assists': 24,
 'baronKills': 0,
 'basicPings': 0,
 'challenges': {'12AssistStreakCount': 0,
  'HealFromMapSources': 2272.523127555847,
  'InfernalScalePickup': 0,
  'SWARM_DefeatAatrox': 0,
  'SWARM_DefeatBriar': 0,
  'SWARM_DefeatMiniBosses': 0,
  'SWARM_EvolveWeapon': 0,
  'SWARM_Have3Passives': 0,
  'SWARM_KillEnemy': 0,
  'SWARM_PickupGold': 0,
  'SWARM_ReachLevel50': 0,
  'SWARM_Survive15Min': 0,
  'SWARM_WinWith5EvolvedWeapons': 0,
  'abilityUses': 245,
  'acesBefore15Minutes': 1,
  'alliedJungleMonsterKills': 0,
  'baronTakedowns': 0,
  'blastConeOppositeOpponentCount': 0,
  'bountyGold': 150,
  'buffsStolen': 0,
  'completeSupportQuestInTime': 0,
  'controlWardsPlaced': 0,
  'damagePerMinute': 1844.1

In [14]:
def winloss(puuid_input,match_data):
    participantid = match_data['metadata']['participants'].index(puuid)
    return match_data['info']['participants'][participantid]['win']



In [15]:
regiont = 'americas'
matcht = match_we_want
api_keyt = riot_key




In [16]:
matchurl_ranked = f"https://americas.api.riotgames.com/lol/match/v5/matches/by-puuid/{puuid}/ids?type=ranked&start=0&count=50&api_key={riot_key}"

matches_ranked_response = requests.get(matchurl_ranked,headers=headers)

matchees_ranked = matches_ranked_response.json()

In [17]:
def get_matches(region,puuid,q_type,count,api_key):
    api_url = f"https://{region}.api.riotgames.com/lol/match/v5/matches/by-puuid/{puuid}/ids?type={q_type}&start=0&count={count}&api_key={api_key}"

    resp = requests.get(api_url)

    return resp.json()




In [18]:
queue = 'ranked'
counter = 50
match_data = get_matches(regiont,puuid,queue,counter,riot_key)

In [19]:
match_data

['NA1_5335539119',
 'NA1_5334986868',
 'NA1_5334960534',
 'NA1_5334022246',
 'NA1_5333891428',
 'NA1_5333881494',
 'NA1_5333584954',
 'NA1_5333522630',
 'NA1_5333501973',
 'NA1_5333490206',
 'NA1_5333475830',
 'NA1_5333148605',
 'NA1_5333114276',
 'NA1_5333094914',
 'NA1_5333057353',
 'NA1_5332810309',
 'NA1_5332762039',
 'NA1_5330924022',
 'NA1_5330919162',
 'NA1_5330914247',
 'NA1_5330910084',
 'NA1_5330315698',
 'NA1_5330312101',
 'NA1_5330306889',
 'NA1_5330297600',
 'NA1_5330292844',
 'NA1_5329038338',
 'NA1_5329032224',
 'NA1_5329022518',
 'NA1_5329010235',
 'NA1_5328999735',
 'NA1_5328984739',
 'NA1_5328330617',
 'NA1_5328317425',
 'NA1_5328205259',
 'NA1_5328190185',
 'NA1_5328147367',
 'NA1_5328123656',
 'NA1_5327631173',
 'NA1_5327518517',
 'NA1_5327494896',
 'NA1_5326618681',
 'NA1_5326594474',
 'NA1_5325703660',
 'NA1_5296914091',
 'NA1_5186020870',
 'NA1_5179169190',
 'NA1_5179143644',
 'NA1_5144202783',
 'NA1_5144184455']

In [ ]:
def get_all_match_data_normalized(match_ids, player_puuid, api_key, region) :
    """
    grabbing the data from each of our matches(last 50) to make a dataframe for further analysis. For now, just grabbing literally everything. 
    I will clean it later or trim the function to grab relevant stuff.
    """
    
    base_url = f"https://{region}.api.riotgames.com/lol/match/v5/matches/"
    headers = {"X-Riot-Token": api_key}
    
    all_player_data = []
    
    for i, match_id in enumerate(match_ids):
        try:
            print(f"Fetching match {i+1}/{len(match_ids)}: {match_id}")
            
            response = requests.get(f"{base_url}{match_id}", headers=headers)
            
            if response.status_code == 200:
                match_json = response.json()
                
                #getting the player we want 
                metadata = match_json['metadata']
                info = match_json['info']
                
                try:
                    participant_index = metadata['participants'].index(player_puuid)
                    player_data = info['participants'][participant_index]
                    
                    player_data['match_id'] = metadata['matchId']
                    player_data['game_duration'] = info['gameDuration']
                    player_data['game_mode'] = info['gameMode']
                    player_data['game_type'] = info['gameType']
                    player_data['queue_id'] = info['queueId']
                    player_data['game_version'] = info['gameVersion']
                    player_data['game_creation'] = info['gameCreation']
                    player_data['game_end'] = info.get('gameEndTimestamp')
                    
                    all_player_data.append(player_data)
                    
                except ValueError:
                    print(f"Player not found in match {match_id}")
                   #i had to add sleeping to this bc i can only do so many requests on trial api key.  
            elif response.status_code == 429:
                print("Rate limited. Waiting 2 seconds...")
                time.sleep(2)
                continue
            else:
                print(f"Error fetching {match_id}: {response.status_code}")
                
        except Exception as e:
            print(f"Exception for match {match_id}: {e}")
            
        time.sleep(0.1)
    
    # normalize the json data to a df 
    if all_player_data:
        df = pd.json_normalize(all_player_data)
        print(f"\nDataFrame created with {len(df)} matches and {len(df.columns)} columns!")
        print(f"Shape: {df.shape}")
        return df
    else:
        print("No match data collected")
        return pd.DataFrame()

In [21]:
our_last_x_games = get_all_match_data_normalized(match_data,puuid,riot_key,regiont)

Fetching match 1/50: NA1_5335539119
Fetching match 2/50: NA1_5334986868
Fetching match 3/50: NA1_5334960534
Fetching match 4/50: NA1_5334022246
Fetching match 5/50: NA1_5333891428
Fetching match 6/50: NA1_5333881494
Fetching match 7/50: NA1_5333584954
Fetching match 8/50: NA1_5333522630
Fetching match 9/50: NA1_5333501973
Fetching match 10/50: NA1_5333490206
Fetching match 11/50: NA1_5333475830
Fetching match 12/50: NA1_5333148605
Fetching match 13/50: NA1_5333114276
Fetching match 14/50: NA1_5333094914
Fetching match 15/50: NA1_5333057353
Fetching match 16/50: NA1_5332810309
Fetching match 17/50: NA1_5332762039
Fetching match 18/50: NA1_5330924022
Fetching match 19/50: NA1_5330919162
Fetching match 20/50: NA1_5330914247
Fetching match 21/50: NA1_5330910084
Fetching match 22/50: NA1_5330315698
Fetching match 23/50: NA1_5330312101
Fetching match 24/50: NA1_5330306889
Fetching match 25/50: NA1_5330297600
Fetching match 26/50: NA1_5330292844
Fetching match 27/50: NA1_5329038338
Fetching m

In [22]:
our_last_x_games.head(15)

PlayerScore0  PlayerScore1  PlayerScore10  PlayerScore11  PlayerScore2  \
0            0.0           0.0            0.0            0.0           0.0   
1            0.0           0.0            0.0            0.0           0.0   
2            0.0           0.0            0.0            0.0           0.0   
3            0.0           0.0            0.0            0.0           0.0   
4            0.0           0.0            0.0            0.0           0.0   
5            0.0           0.0            0.0            0.0           0.0   
6            0.0           0.0            0.0            0.0           0.0   
7            0.0           0.0            0.0            0.0           0.0   
8            0.0           0.0            0.0            0.0           0.0   
9            0.0           0.0            0.0            0.0           0.0   
10           0.0           0.0            0.0            0.0           0.0   
11           0.0           0.0            0.0            0.0           0.0   
12           0.0           0.0            0.0            0.0           0.0   
13           0.0           0.0            0.0            0.0           0.0   
14           0.0           0.0            0.0            0.0           0.0   

    PlayerScore3  PlayerScore4  PlayerScore5  PlayerScore6  PlayerScore7  ...  \
0            0.0           0.0           0.0           0.0           0.0  ...   
1            0.0           0.0           0.0           0.0           0.0  ...   
2            0.0           0.0           0.0           0.0           0.0  ...   
3            0.0           0.0           0.0           0.0           0.0  ...   
4            0.0           0.0           0.0           0.0           0.0  ...   
5            0.0           0.0           0.0           0.0           0.0  ...   
6            0.0           0.0           0.0           0.0           0.0  ...   
7            0.0           0.0           0.0           0.0           0.0  ...   
8            0.0           0.0           0.0           0.0           0.0  ...   
9            0.0           0.0           0.0           0.0           0.0  ...   
10           0.0           0.0           0.0           0.0           0.0  ...   
11           0.0           0.0           0.0           0.0           0.0  ...   
12           0.0           0.0           0.0           0.0           0.0  ...   
13           0.0           0.0           0.0           0.0           0.0  ...   
14           0.0           0.0           0.0           0.0           0.0  ...   

    challenges.soloTurretsLategame  challenges.teleportTakedowns  \
0                              NaN                           NaN   
1                              NaN                           NaN   
2                              NaN                           NaN   
3                              1.0                           NaN   
4                              NaN                           NaN   
5                              NaN                           NaN   
6                              NaN                           1.0   
7                              NaN                           NaN   
8                              NaN                           NaN   
9                              NaN                           NaN   
10                             NaN                           1.0   
11                             NaN                           NaN   
12                             NaN                           NaN   
13                             NaN                           NaN   
14                             NaN                           NaN   

    challenges.baronBuffGoldAdvantageOverThreshold  \
0                                              NaN   
1                                              NaN   
2                                              NaN   
3                                              NaN   
4                                              NaN   
5                                            

In [ ]:
challenge_cols = [col for col in our_last_x_games.columns if 'challenge' in col.lower() or 'playerscore' in col.lower()]

df_clean = our_last_x_games.drop(columns=challenge_cols)


Dropping 165 challenge-related columns


In [24]:
df_clean.head()

allInPings  assistMePings  assists  baronKills  basicPings  \
0           0              0        6           0           0   
1           0              0       11           0           0   
2           0              0       11           0           0   
3           0              0        2           0           0   
4           0              0        5           0           0   

   champExperience  champLevel  championId championName  championTransform  \
0            18179          17         901      Smolder                  0   
1            18072          17         145        Kaisa                  0   
2            17165          17          15        Sivir                  0   
3            12312          14         122       Darius                  0   
4            13656          15          86        Garen                  0   

   ...     game_type  queue_id    game_version  game_creation       game_end  \
0  ...  MATCHED_GAME       420  15.15.699.7589  1753926868320  1753929024776   
1  ...  MATCHED_GAME       420  15.14.697.2104  1753853863804  1753856100743   
2  ...  MATCHED_GAME       420  15.14.697.2104  1753851301759  1753853398760   
3  ...  MATCHED_GAME       420  15.14.697.2104  1753729138808  1753730363970   
4  ...  MATCHED_GAME       420  15.14.697.2104  1753690400341  1753692034054   

   perks.statPerks.defense  perks.statPerks.flex  perks.statPerks.offense  \
0                     5011                  5008                     5005   
1                     5011                  5008                     5005   
2                     5011                  5008                     5005   
3                     5001                  5008                     5005   
4                     5013                  5008                     5008   

                                        perks.styles  bountyLevel  
0  [{'description': 'primaryStyle', 'selections':...          NaN  
1  [{'description': 'primaryStyle', 'selections':...          NaN  
2  [{'description': 'primaryStyle', 'selections':...          NaN  
3  [{'description': 'primaryStyle', 'selections':...          NaN  
4  [{'description': 'primaryStyle', 'selections':...          NaN  

[5 rows x 142 columns]

In [25]:
df_clean.columns

Index(['allInPings', 'assistMePings', 'assists', 'baronKills', 'basicPings',
       'champExperience', 'champLevel', 'championId', 'championName',
       'championTransform',
       ...
       'game_type', 'queue_id', 'game_version', 'game_creation', 'game_end',
       'perks.statPerks.defense', 'perks.statPerks.flex',
       'perks.statPerks.offense', 'perks.styles', 'bountyLevel'],
      dtype='object', length=142)

In [26]:
all_columns = df_clean.columns.tolist()
print(f"Total columns: {len(all_columns)}")
for i, col in enumerate(all_columns):
    print(f"{i+1:3d}. {col}")

Total columns: 142
  1. allInPings
  2. assistMePings
  3. assists
  4. baronKills
  5. basicPings
  6. champExperience
  7. champLevel
  8. championId
  9. championName
 10. championTransform
 11. commandPings
 12. consumablesPurchased
 13. damageDealtToBuildings
 14. damageDealtToObjectives
 15. damageDealtToTurrets
 16. damageSelfMitigated
 17. dangerPings
 18. deaths
 19. detectorWardsPlaced
 20. doubleKills
 21. dragonKills
 22. eligibleForProgression
 23. enemyMissingPings
 24. enemyVisionPings
 25. firstBloodAssist
 26. firstBloodKill
 27. firstTowerAssist
 28. firstTowerKill
 29. gameEndedInEarlySurrender
 30. gameEndedInSurrender
 31. getBackPings
 32. goldEarned
 33. goldSpent
 34. holdPings
 35. individualPosition
 36. inhibitorKills
 37. inhibitorTakedowns
 38. inhibitorsLost
 39. item0
 40. item1
 41. item2
 42. item3
 43. item4
 44. item5
 45. item6
 46. itemsPurchased
 47. killingSprees
 48. kills
 49. lane
 50. largestCriticalStrike
 51. largestKillingSpree
 52. largest

In [27]:
columns_to_keep = ['riotIdGameName','riotIdTagline','profileIcon','win','role','lane','championName','champLevel','kills','deaths','assists','totalDamageDealtToChampions','totalDamageShieldedOnTeammates','magicDamageDealtToChampions','physicalDamageDealtToChampions','trueDamageDealtToChampions','totalHeal','visionScore','visionWardsBoughtInGame','wardsKilled','wardsPlaced','objectivesStolen','totalDamageTaken','totalHeal','timeCCingOthers','totalTimeCCDealt','totalMinionsKilled','consumablesPurchased','damageDealtToBuildings','damageDealtToObjectives','damageDealtToTurrets','dangerPings','allInPings','assistMePings','basicPings','enemyMissingPings','enemyVisionPings','onMyWayPings','pushPings','dragonKills','firstBloodAssist','firstTowerAssist','firstTowerKill','gameEndedInSurrender','goldEarned','goldSpent','inhibitorKills','game_duration','neutralMinionsKilled','placement','item0','item1','item2','item3','item4','item5','item6']
new_test_df = df_clean[columns_to_keep]

new_test_df.head(10)

riotIdGameName riotIdTagline  profileIcon    win     role    lane  \
0       BANGBANG          7999         6852   True    CARRY  BOTTOM   
1       BANGBANG          7999         6852  False    CARRY  BOTTOM   
2       BANGBANG          7999         6852   True    CARRY  BOTTOM   
3       BANGBANG          7999         6852   True     SOLO     TOP   
4       BANGBANG          7999         6852  False     SOLO     TOP   
5       BANGBANG          7999         6852  False     SOLO  MIDDLE   
6       BANGBANG          7999         6852  False     SOLO  MIDDLE   
7       BANGBANG          7999         6852   True     SOLO  MIDDLE   
8       BANGBANG          7999         6852  False    CARRY  BOTTOM   
9       BANGBANG          7999         6852  False  SUPPORT    NONE   

  championName  champLevel  kills  deaths  ...  game_duration  \
0      Smolder          17     13       4  ...           2065   
1        Kaisa          17      9       8  ...           2213   
2        Sivir          17      8       5  ...           2077   
3       Darius          14      5       1  ...           1205   
4        Garen          15      2       5  ...           1613   
5        Annie          13      2       8  ...           1666   
6       Syndra          16      6       4  ...           1809   
7        Annie          18     15       6  ...           2682   
8        Kaisa          16     13       7  ...           2098   
9          Lux           7      1       2  ...            930   

   neutralMinionsKilled  placement  item0  item1  item2  item3  item4  item5  \
0                    19          0   3161   3508      0   3035   3171   3031   
1                    16          0   3072   6672   3124   3115   3006   1058   
2                    14          0   3033   3032   3031   3172   6675      0   
3                     8          0   1054   3078      0   3047   1037   2021   
4                     6          0   1054   6631   1036   3071   3006      0   
5                     0          0   1056   3118   3020   4646   1052      0   
6                     1          0   1056   2503   4629   1058      0   3020   
7                     8          0   3135   3118   3113   3020   3089   4645   
8                    35          0   3072   6672   3124   3172   3115   3302   
9                     0          0   6655   3871   2055      0      0      0   

   item6  
0   3364  
1   3364  
2   3340  
3   3340  
4   3340  
5   3340  
6   3340  
7   3364  
8   3340  
9   3364  

[10 rows x 57 columns]

In [29]:
new_test_df_2 = new_test_df.copy()
new_test_df_2['total_farm'] = new_test_df['totalMinionsKilled'] + new_test_df['neutralMinionsKilled']

new_test_df_2['cs_per_min'] = (new_test_df_2['total_farm'] / (new_test_df['game_duration']/60)).round(2)

new_test_df_2['game_time_mins'] = (new_test_df['game_duration'] /60).round(2)

new_test_df_2['kda'] = (new_test_df_2['kills'] + new_test_df_2['assists'] / new_test_df_2['deaths']).round(2)

new_test_df_2['gold_per_min'] = (new_test_df_2['goldEarned'] / new_test_df_2['game_time_mins']).round(2)

new_test_df_2['dmg_per_min'] = (new_test_df_2['totalDamageDealtToChampions'] / new_test_df_2['game_time_mins']).round(2)



new_test_df_2.head(15)

riotIdGameName riotIdTagline  profileIcon    win     role    lane  \
0        BANGBANG          7999         6852   True    CARRY  BOTTOM   
1        BANGBANG          7999         6852  False    CARRY  BOTTOM   
2        BANGBANG          7999         6852   True    CARRY  BOTTOM   
3        BANGBANG          7999         6852   True     SOLO     TOP   
4        BANGBANG          7999         6852  False     SOLO     TOP   
5        BANGBANG          7999         6852  False     SOLO  MIDDLE   
6        BANGBANG          7999         6852  False     SOLO  MIDDLE   
7        BANGBANG          7999         6852   True     SOLO  MIDDLE   
8        BANGBANG          7999         6852  False    CARRY  BOTTOM   
9        BANGBANG          7999         6852  False  SUPPORT    NONE   
10       BANGBANG          7999         6852  False      DUO    NONE   
11       BANGBANG          7999         6852  False     SOLO  MIDDLE   
12       BANGBANG          7999         6852  False     SOLO  MIDDLE   
13       BANGBANG          7999         6852  False      DUO    NONE   
14       BANGBANG          7999         6852   True      DUO  MIDDLE   

   championName  champLevel  kills  deaths  ...  item3  item4  item5  item6  \
0       Smolder          17     13       4  ...   3035   3171   3031   3364   
1         Kaisa          17      9       8  ...   3115   3006   1058   3364   
2         Sivir          17      8       5  ...   3172   6675      0   3340   
3        Darius          14      5       1  ...   3047   1037   2021   3340   
4         Garen          15      2       5  ...   3071   3006      0   3340   
5         Annie          13      2       8  ...   4646   1052      0   3340   
6        Syndra          16      6       4  ...   1058      0   3020   3340   
7         Annie          18     15       6  ...   3020   3089   4645   3364   
8         Kaisa          16     13       7  ...   3172   3115   3302   3340   
9           Lux           7      1       2  ...      0      0      0   3364   
10        Annie          11      2       3  ...   3020      0      0   3340   
11          Lux          13      4       2  ...   3020   6655      0   3340   
12        Annie          16      5       6  ...   3089      0   3020   3340   
13        Annie          10      0       4  ...   3020      0      0   3340   
14        Annie          18      8      11  ...   4630   4645   3118   3364   

    total_farm  cs_per_min  game_time_mins    kda  gold_per_min  dmg_per_min  
0          235        6.83           34.42  14.50        466.73       857.44  
1          265        7.18           36.88  10.38        464.80      1211.90  
2          254        7.34           34.62  10.20        440.24       867.62  
3          159        7.92           20.08   7.00        445.27       577.79  
4          196        7.29           26.88   3.00        331.81       508.97  
5          120        4.32           27.77   3.00        285.31       511.20  
6          182        6.04           30.15   8.00        354.39       719.64  
7          237        5.30           44.70  17.50        396.17       892.10  
8          250        7.15           34.97  13.86        488.59      1136.03  
9           22        1.42           15.50   1.00        235.94       412.32  
10         117        6.21           18.83   2.33        308.98       639.46  
11         152        7.46           20.38   5.00        443.38       719.58  
12         195        5.82           33.52   5.50        353.61       687.20  
13         111        7.16           15.50   0.75        305.23       333.23  
14         158        4.79           33.00   8.45        387.00       896.39  

[15 rows x 63 columns]

To recap, so far we have:

1. Grabbed our selected player's puuid, which we can use to get their recent matches
2. Wrote functions to grab our recent matches, with the amount being variable
3. We take that whole list of matches, with the queue type being changable, and grabbed pretty much all of the json data for each of those matches, and added it all to a dataframe. So for this, I didnt really want to specify certain values to grab because its possible in the future I want to grab different stuff, etc, and i only know how to grab it all, then noramlize it, which is what I did.
4. Cleaned the shit out of it originally, trimmed the columns down to avoid getting 142 challegne cols, lol. 
5. Added some of my own derived columns, like cs/min, gold/min, kda, gametime in mins, etc. We had to make a total farm col too bc the original data doesnt count jungle camps in our cs score. 



Whats next:

1. finalize the data I want to grab and get it cleaned and prepared for analysis
2. Set up an automated way to get and clean the data whenever we have a player we want to look up
3. do some data analysis to actually get some info out of this data
4. Make a dashboard using Dash, which ive only briefly used. 
5. Id like to add more derived columns or another table that can show us our champ winrate, winrate vs this champ, etc. 

Way down the line:

1. Establish a database to store player/game data so we dont have to hit the api everytime we want to get some game data
2. Player performance comparison to see if we can see the difference between player ranks purely based on game stats. 